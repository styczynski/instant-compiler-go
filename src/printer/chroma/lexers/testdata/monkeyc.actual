using Toybox.Application as App;
using Toybox.System;

const PI = 3.14;

class MyProjectApp extends App.AppBase {
    protected var y;

    function initialize () {
        me.y = "Hello";
        self.y = "World";
        var x = add( 3, 4 );
        var array = new [x];

        // Initialize the sub-arrays
        for( var i = 0; i < x; i += 1 ) {
            array[i] = new [5];
        }

        var dict = { "a" => 1, "b" => 2 };
        var person = { :firstName=>"Bob", :lastName=>"Jones" };
    }

    public function onStart(state) {
        var v = new Foo();
        var m = v.method(:op);
        m.invoke(1,2l);
    }

    function getInitialView() {
        return [ new MyProjectView() ];
    }

    function add( a, b ) {
        return a + b;
    }

    function thisFunctionUsesAdd() {
        var a = add( 1, 0x03f ); // Return  4
        var b = add( "Hello ", "World" ); // Returns "Hello World"
    }
    
}

class Foo  {
    function op(a, b) {}
}