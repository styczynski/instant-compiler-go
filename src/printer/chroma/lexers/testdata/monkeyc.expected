[
  {"type":"KeywordNamespace","value":"using"},
  {"type":"Text","value":" "},
  {"type":"NameNamespace","value":"Toybox.Application"},
  {"type":"Text","value":" "},
  {"type":"KeywordNamespace","value":"as"},
  {"type":"Text","value":" "},
  {"type":"NameNamespace","value":"App"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordNamespace","value":"using"},
  {"type":"Text","value":" "},
  {"type":"NameNamespace","value":"Toybox.System"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordType","value":"const"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"PI"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberFloat","value":"3.14"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"class"},
  {"type":"Text","value":" "},
  {"type":"NameClass","value":"MyProjectApp"},
  {"type":"Text","value":" "},
  {"type":"KeywordDeclaration","value":"extends"},
  {"type":"Text","value":" "},
  {"type":"NameClass","value":"App.AppBase"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n    "},
  {"type":"KeywordType","value":"protected"},
  {"type":"Text","value":" "},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"y"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n    "},
  {"type":"KeywordDeclaration","value":"function"},
  {"type":"Text","value":" "},
  {"type":"NameFunctionMagic","value":"initialize"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"()"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n        "},
  {"type":"NameBuiltinPseudo","value":"me"},
  {"type":"Punctuation","value":"."},
  {"type":"Name","value":"y"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"Hello\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n        "},
  {"type":"NameBuiltinPseudo","value":"self"},
  {"type":"Punctuation","value":"."},
  {"type":"Name","value":"y"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"World\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n        "},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"x"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"add"},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"3"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"4"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":"\n        "},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"array"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"OperatorWord","value":"new"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"["},
  {"type":"Name","value":"x"},
  {"type":"Punctuation","value":"];"},
  {"type":"Text","value":"\n\n        "},
  {"type":"CommentSingle","value":"// Initialize the sub-arrays\n"},
  {"type":"Text","value":"        "},
  {"type":"Keyword","value":"for"},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":" "},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"i"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"0"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"i"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"\u003c"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"x"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"i"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"+="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n            "},
  {"type":"Name","value":"array"},
  {"type":"Punctuation","value":"["},
  {"type":"Name","value":"i"},
  {"type":"Punctuation","value":"]"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"OperatorWord","value":"new"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"["},
  {"type":"LiteralNumberInteger","value":"5"},
  {"type":"Punctuation","value":"];"},
  {"type":"Text","value":"\n        "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n        "},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"dict"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"a\""},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=\u003e"},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"b\""},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=\u003e"},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"2"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"};"},
  {"type":"Text","value":"\n        "},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"person"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringSymbol","value":":firstName"},
  {"type":"Operator","value":"=\u003e"},
  {"type":"LiteralStringDouble","value":"\"Bob\""},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"LiteralStringSymbol","value":":lastName"},
  {"type":"Operator","value":"=\u003e"},
  {"type":"LiteralStringDouble","value":"\"Jones\""},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"};"},
  {"type":"Text","value":"\n    "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n    "},
  {"type":"KeywordType","value":"public"},
  {"type":"Text","value":" "},
  {"type":"KeywordDeclaration","value":"function"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"onStart"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"state"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n        "},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"v"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"OperatorWord","value":"new"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"Foo"},
  {"type":"Punctuation","value":"();"},
  {"type":"Text","value":"\n        "},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"m"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"v"},
  {"type":"Punctuation","value":"."},
  {"type":"NameBuiltin","value":"method"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralStringSymbol","value":":op"},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":"\n        "},
  {"type":"Name","value":"m"},
  {"type":"Punctuation","value":"."},
  {"type":"Name","value":"invoke"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":","},
  {"type":"LiteralNumberInteger","value":"2l"},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":"\n    "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n    "},
  {"type":"KeywordDeclaration","value":"function"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"getInitialView"},
  {"type":"Punctuation","value":"()"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n        "},
  {"type":"Keyword","value":"return"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"["},
  {"type":"Text","value":" "},
  {"type":"OperatorWord","value":"new"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"MyProjectView"},
  {"type":"Punctuation","value":"()"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"];"},
  {"type":"Text","value":"\n    "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n    "},
  {"type":"KeywordDeclaration","value":"function"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"add"},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":" "},
  {"type":"Name","value":"a"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"Name","value":"b"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n        "},
  {"type":"Keyword","value":"return"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"+"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"b"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n    "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n    "},
  {"type":"KeywordDeclaration","value":"function"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"thisFunctionUsesAdd"},
  {"type":"Punctuation","value":"()"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n        "},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"add"},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberHex","value":"0x03f"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// Return  4\n"},
  {"type":"Text","value":"        "},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"b"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"add"},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"Hello \""},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"World\""},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// Returns \"Hello World\"\n"},
  {"type":"Text","value":"    "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n    \n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"class"},
  {"type":"Text","value":" "},
  {"type":"NameClass","value":"Foo"},
  {"type":"Text","value":"  "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n    "},
  {"type":"KeywordDeclaration","value":"function"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"op"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"a"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"Name","value":"b"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{}"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"}
]
